python/cuml/dask/neighbors/kneighbors_regressor.py

```diff
@@ -198,10 +198,15 @@ def predict(self, X, convert_dtype=True):
 
         comms.destroy()
 
-        out = to_output(out_futures, self.datatype)
-        out_i = to_output(out_i_futures, self.datatype)
-        out_d = to_output(out_d_futures, self.datatype)
-        return out, out_i, out_d
+        out = to_output(out_futures, self.datatype).squeeze()
+
+        out_i = to_output(out_i_futures, self.datatype)  # noqa: F841
+        out_d = to_output(out_d_futures, self.datatype)  # noqa: F841
+
+        # Returning predictions only to conform with Scikit-Learn behavior
+        # Complete removal of indices and distances will be effective
```

Comment:
I'm not sure we need to completely remove the indices and distances, unless you see a major benefit to doing so. I'm in favor of removing this comment altogether, but if you think it should stay, we should probably create a Github issue for it and reference here in a TODO (along with the same comment in the classifier above).

Reply:
I think that it does not create major performance issue as long as these Dask arrays are not gathered back to the client. However, it creates some unnecessary allocations on the workers, but I don't think that it's a big issue for now. I think that it's preferable to update this code along with the merge of MNMG NN with MNMG NN Cl&Re. I will open an issue for this.